src/hdrgen.d

```diff
@@ -1742,10 +1742,16 @@ public:
         //printf("FuncDeclaration::toCBuffer() '%s'\n", f.toChars());
         if (stcToBuffer(buf, f.storage_class))
             buf.writeByte(' ');
-        typeToBuffer(f.type, f.ident);
+        auto tf = cast(TypeFunction)f.type;
+        typeToBuffer(tf, f.ident);
         if (hgs.hdrgen == 1)
         {
-            if (f.storage_class & STCauto)
+            if(f.fbody && f.fbody.toChars()[0] == '\0')
```

Comment:
@RazvanN7 is right.
In case of an empty body it does actually not matter if the type was auto or not.
Since the only thing an `{ }` body can evaluate to is `void` regardless.